### ISSUE TITLE
aws_s3_bucket region ignored, provider region takes precedence

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#592]
_This issue was originally opened by @mzraly as hashicorp/terraform#12512. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
Terraform v0.8.8

### Affected Resource(s)
- aws_s3_bucket

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-west-2"
}

resource "aws_s3_bucket" "res_s3_bucket" {
  bucket = "tf-bucket-region-issue"
  region = "us-east-1"
}
```

### Debug Output
https://gist.github.com/mzraly/6c4cbdd2f8016c0526bd3b18b7a43a01

### Panic Output
None

### Expected Behavior
Documentation at https://www.terraform.io/docs/providers/aws/r/s3_bucket.html suggests that by specifying the region in an aws_s3_bucket resource that bucket will be created in that region.  In the example above, we expected the bucket to be created in us-east-1.

### Actual Behavior
The bucket is created in the region specified by the provider, us-west-2.

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
None

### References
None I am aware of.

**Comment by bflad** [comment]
See also: #1923

We should remove `Optional: true` from the `region` attribute of the `aws_s3_bucket` resource in a major version release.

**Comment by wperron** [comment]
hey is anyone working on this issue? I'd be willing to give it a shot

**Comment by bflad** [comment]
Removing the confusing configurability of the `region` attribute in the `aws_s3_bucket` resource has been merged and will release with version 3.0.0 of the Terraform AWS Provider, likely in about two weeks. In the future if we want to allow individual resource region configuration, we will need to come up with a design proposal since it will be very high impact on the overall user experience of the provider.
